Explain Operator Reconciliation (#2431)
TykTechnologies/tyk-docs: tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-reconciliation.md

## When would reconciliation happen?
Before diving into Tyk Operator reconciliation, let me briefly mention some technical details about how and when reconciliation happens. Reconciliation only happens when certain events happen on your cluster or objects. Therefore, Reconciliation will **NOT** be triggered when there is an update or modification on Tyk’s side. It only watches certain Kubernetes events and is triggered based on them. Usually, the reconciliation happens when you modify a Kubernetes object or when the cache used by the controller expires - side note, controllers, in general, use cached objects to reduce the load in the Kube API server. Typically, caches expire in ~10 hours or so but the expiration time might change based on Operator configurations.

So, in order to trigger Reconciliation, you can either
- modify an object, which will trigger reconciliation over this modified object or,
- restart Tyk Operator pod, which will trigger reconciliation over each of the objects watched by Tyk Operator.

## What happens during Reconciliation?

> Tyk Operator will compare desired state of the Kubernetes object with the observed state in Tyk. If there are drift, Tyk Operator will update the actual state on Tyk with the desired state. In the reconciliation, we mainly control three operations; DELETE, CREATE, and UPDATE.

Reviewer: ```suggestion
Tyk Operator will compare desired state of the Kubernetes object with the observed state in Tyk. If there is a drift, Tyk Operator will update the actual state on Tyk with the desired state. In the reconciliation, Tyk Operator mainly controls three operations; DELETE, CREATE, and UPDATE.
```

- **CREATE** - an object is created in Kubernetes but not exists in Tyk
- **UPDATE** - an object is in different in Kubernetes and Tyk (we compare that by hash)
- **DELETE** - an object is deleted in Kubernetes but exists in Tyk

### Drift Detection
If human operators or any other system delete or modify ApiDefinition from Tyk Gateway or Dashboard, Tyk Operator will restore the desired state back to Tyk during reconciliation. This is called Drift Detection. It can protect your systems from unauthorised or accidental modifications. It is a best practice to limit user access rights on production environment to read-only in order to prevent accidental updates through API Manager directly.

### Start using Tyk Operator
If you are already running Tyk for your APIs, you can convert existing APIs and Policies into CRDs and let Tyk Operator manages the state of it going forward. Check out our [Migration Guide](https://tyk.io/docs/tyk-stack/tyk-operator/migration/).

> It is also useful to setup your CI/CD pipeline to automate changes of API Definitions in Git to be published to Tyk. See [Using Tyk Operator to enable GitOps with Tyk](https://tyk.io/docs/getting-started/key-concepts/gitops-with-tyk/) for the key concepts behind.

Reviewer 2: @caroltyk per our guide we need to use "ref" for internal links
